packages/components/pie-button/src/index.ts

```diff
@@ -48,7 +57,7 @@ export class PieButton extends LitElement {
                 size=${size}
                 ?disabled=${disabled}
                 ?isFullWidth=${isFullWidth}>
-                <slot></slot>
+                ${isLoading ? this.renderLoadingSpinner() : html`<slot></slot>`}
```

Comment:
We can make use of the `until` directive here: https://lit.dev/docs/templates/directives/#until

Reply:
I've removed this now – after speaking to design, they wanted to keep the height/width of the button dependent on the text in the button, so rather than rendering the loading spinner as markup, I changed it to add it as a pseudo-element (so this isn't needed now)